buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/PerformanceTest.java

```diff
@@ -18,105 +18,204 @@
 
 import org.gradle.api.internal.tasks.options.Option;
 import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.Nested;
 import org.gradle.api.tasks.Optional;
+import org.gradle.api.tasks.Internal;
 import org.gradle.api.tasks.OutputDirectory;
 import org.gradle.gradlebuild.test.integrationtests.DistributionTest;
+import org.gradle.api.specs.Spec;
+import org.gradle.api.Task;
+
+import org.gradle.api.tasks.CacheableTask;
 
 import javax.annotation.Nullable;
 import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.Map.Entry;
+import javax.inject.Inject;
+
+import org.gradle.process.CommandLineArgumentProvider;
 
 /**
  * A test that checks execution time and memory consumption.
  */
+@CacheableTask
 public class PerformanceTest extends DistributionTest {
-
+    private String scenarios;
     private String baselines;
     private String warmups;
     private String runs;
     private String checks;
     private String channel;
+    private String resultStoreClass = "org.gradle.performance.results.AllResultsStore";
+    private boolean flamegraphs;
+
+    protected Map<String, String> databaseParameters = new HashMap<>();
```

Comment:
Why is this map protected and not `private final` with a getter? Note that the URL to the database is an input to the task.